### ISSUE TITLE
Missing type definition / interface for standalone bundle

### ISSUE THREAD

**Issue by Ketec** [josdejong/svelte-jsoneditor#91]
```
Error: node_modules/svelte-jsoneditor/components/JSONEditor.svelte.d.ts:37:31 - error TS2304: Cannot find name 'OptionalJSONEditorProps'.

37         updateProps?: (props: OptionalJSONEditorProps) => void;

```
v 0.3.58

I see it exists in /src/lib/components/JSONEditor.svelte - but not in the bundle installed with npm.

This ends up as a fatal error in strongly typed languages like Angular.

https://stackblitz.com/edit/base-angular-12-app-yvpgys?file=src/app/app.component.ts (fails because the whole definition is not compiling)

For a local copy, I could edit node modules definitions and set these to `any` to test - it compiles and runs (but with issues in the tree view - like update not refreshing view with new data and clicking in editor after update breaks with update async/undefined error ).

**Comment by josdejong** [comment]
Thanks for reporting. This interface was exported from a Svelte component. Apparently that doesn't work. I've moved the interface to a separate .ts file to solve it.

> For a local copy, I could edit node modules definitions and set these to any to test - it compiles and runs (but with issues in the tree view - like update not refreshing view with new data and clicking in editor after update breaks with update async/undefined error ).

Type definitions cannot alter the actual JavaScript behavior, so there is probably an other reason for the issue you mention.

**Comment by josdejong** [comment]
Should be fixed now in `v0.3.59`

**Comment by josdejong** [comment]
I did some debugging. The types are there, but right now the types rely on types defined in the `svelte` package, which is not installed by default. If you install `svelte`, all works fine. See https://github.com/josdejong/svelte-jsoneditor/issues/19#issuecomment-1147184932, help would be welcome to fix this.
